### PR TITLE
Setup redux slices and store

### PR DESCRIPTION
**Pull request by preciousuwhubetine**
In this branch:
- I installed the `@reduxjs/toolkit` and `react-redux` packages
- I set up the redux store using the `configureStore` method
- I created two slices: `booksSlice` and `categoriesSlice`
- The `booksSlice` manages state for the books feature of the site
- The `categoriesSlice` manages state for the categories feature of the site

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    removeBook: (state, action) => ({
+      ...state,
+      books: state.books.filter(
+        (book) => book.id !== action.payload,
+      ),
+    }),
+  },
+});
+
+export default booksSlice.reducer;
```
Comment (bobb-Rob) [inline src/redux/books/booksSlice.js]: Well done @preciousbetine 

You did everything well except that I do not see how you exported the actions using the ES6 object destructuring from `booksSlice.actions`. 

```javascript
export const { addBook, removeBook } = booksSlice.actions
```

Kindly add it

File: src/redux/categories/categoriesSlice.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {
+    checkStatus: () => 'Under construction',
+  },
+});
+
+export default categoriesSlice.reducer;
```
Comment (bobb-Rob) [inline src/redux/categories/categoriesSlice.js]: Well done @preciousbetine 

You did everything well except that I do not see how you exported the actions using the ES6 object destructuring from ``categoriesSlice.actions``. 

```javascript
  export const { checkStatus } = categoriesSlice.actions
```

Kindly add it

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️
Hi @preciousbetine , 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Highlights
✅ No Linter error
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)
✅ Neat Commit messages

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categoriesSlice.js (hunk 2 above)
Comment (preciousuwhubetine) [reply src/redux/categories/categoriesSlice.js]: Hi @bobb-Rob , thanks for taking the time to review my work!  Just to clarify, exporting the slice's actions wasn't explicitly stated as a requirement for this project. Nonetheless, I will add it and re-request a review.

**Review by prolajumokeoni — APPROVED** [review]
Hi @preciousbetine 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
  - Changes requested have been implemented ✅ 
  

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- It would be better to use create-react-app command for initializing the project, this was a requirement for the previous milestone, please always follow the requirements in future milestones 👍🏿  

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categoriesSlice.js (hunk 2 above)
Comment (bobb-Rob) [reply src/redux/categories/categoriesSlice.js]: Hi @preciousbetine, you did well adding it all the same. It was explicitly stated, please see below 👇🏾 

image

Comment (preciousuwhubetine) [reply src/redux/categories/categoriesSlice.js]: @bobb-Rob Thanks. Must have been an oversight

**Pull request merged**